### PR TITLE
Feature/exe 1884 make community detection deterministic

### PR DESCRIPTION
**Pull request by ptajvar**
Setting a random seed for leiden algorithm so that component IDs remain constant through multiple runs of pipeline.

Fixes: #EXE-1884

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
## PR checklist:

- [x] This comment contains a description of changes (with reason).
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by johandahlberg — APPROVED** [review]
This looks good to me. Great work!

File: src/pixelator/annotate/__init__.py [hunk 1]
```diff
@@ -221,7 +221,7 @@ def annotate_components(
     # components clustering
     if adata.n_obs > MINIMUM_NBR_OF_CELLS_FOR_ANNOTATION:
         # perform the unsupervised clustering
-        cluster_components(adata=adata, inplace=True)
+        cluster_components(adata=adata, inplace=True, random_seed=1)
```
Comment (ambarrio) [inline src/pixelator/annotate/__init__.py:224]: why the different random_seed here and in _networkx?

File: src/pixelator/graph/backends/implementations/_networkx.py [hunk 2]
```diff
@@ -305,6 +305,7 @@ def community_leiden(
             graph,
             use_modularity=True,
             randomness=beta,
+            random_seed=42,
```
Comment (ambarrio) [inline src/pixelator/graph/backends/implementations/_networkx.py:308]: Should we use the same see everywhere? i.e. `1477` :)

**Review by ambarrio — APPROVED** [review]
A couple of comments but nothing major. Please, read my comments on the ticket.

**Comment by ptajvar** [comment]
Superseded by [PR 188](https://github.com/PixelgenTechnologies/pixelator/pull/188).

**Pull request closed without merging**